### PR TITLE
PTDF Boundaries with EI Code instead of Country Code

### PR DESCRIPTION
**Pull request by baptsegu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
New feature

**What is the current behavior?** *(You can also link to an open issue here)*
The parameter ptdfBoundaries was defined as a list of **Country** pairs. Its importers use 2-characters country codes.

**What is the new behavior (if this is a feature change)?**
The parameter ptdfBoundaries is defined as a list of **EICode** pairs. Its importers use 2-characters country codes OR 16-characters EICode.

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

YES !

First breaking change : 

- The enum `EICode` from package _util_ has been renamed `CountryEICode` and moved to package _common_
- The class `ReferenceProgramArea` from package _reference-program_ has been renamed `EICode` and moved to package _common_.

The class called `EICode` therefore now handle any EICode, EICode from countries, but also non-country (e.g. virtual hubs, like Alegro)

Second breaking change : 

* the parameter `max-number-of-boundaries-for-skipping-network-actions` was previously identifying the proximity of countries with ptdf-boundaries parameter.
* the parameter `max-number-of-boundaries-for-skipping-network-actions` is now identifying the proximity of countries by looking in the network (ptdf-boundaries cannot be used anymore as it contains EICode and not Country)

Third breaking change : 

The parameter `ptdf-boundaries` has been remaned `relative-margin-ptdf-boundaries`, to avoid confusion with other possible use of the PTDFs (exemple : loop-flow computation)

Fourth breaking change : 

In json or yml config files, the boundaries of `relative-margin-ptdf-boundaries` has to be defined with "code1/code2". A "/" is used instead of "-" as a separator as EICode usually already contains "-".

### CODE REVIEW

File: commons/src/main/java/com/farao_community/farao/commons/CountryGraph.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public class CountryGraph {
```
[the remark this answers is not in the document]
Comment (phiedw) [reply commons/src/main/java/com/farao_community/farao/commons/CountryGraph.java:22]: I like the name CountryGraph. I think it's more explicit about what this class does: checking if two countries share a boundary.

[the remark this answers is not in the document]
Comment (pet-mit) [reply commons/src/main/java/com/farao_community/farao/commons/CountryGraph.java:22]: ok ok

[the remark this answers is not in the document]
Comment (baptsegu) [reply commons/src/main/java/com/farao_community/farao/commons/CountryGraph.java:22]: I keept CountryGraph then :p

File: commons/src/main/java/com/farao_community/farao/commons/CountryGraph.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public boolean areNeighbors(Country country1, Country country2, int maxNumberOfBoundaries) {
+        if (country1.equals(country2)) {
+            return true;
+        }
+        if (maxNumberOfBoundaries <= 0) {
+            return false;
+        }
+
+        for (CountryBoundary boundary : boundaries) {
+            Pair<Country, Country> countryPair = boundary.getCountryPair();
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply commons/src/main/java/com/farao_community/farao/commons/CountryGraph.java:70]: You're right, I changed that.

**Review by pet-mit — APPROVED** [review]

**Pull request merged**